From 05d0012798d3421b1fe59dfd2cc26f427697304a Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Wed, 9 Oct 2013 11:15:25 +0530
Subject: [PATCH 101/121] Removed commented code in addb_pfom.c thats posts
 monitor summaries, since it was failing UT. Added a condition in posting that
 seems appropriate due to which existing ADDB UTs pass.

---
 addb/addb_pfom.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/addb/addb_pfom.c b/addb/addb_pfom.c
index f8fe4bf..95612a4 100644
--- a/addb/addb_pfom.c
+++ b/addb/addb_pfom.c
@@ -206,7 +206,7 @@ static int addb_pfom_fo_tick(struct m0_fom *fom)
 	struct m0_reqh_service *rsvc = &svc->as_reqhs;
 	int                     rc = M0_FSO_AGAIN;
 	m0_time_t               now;
-//	int                     err;
+	int                     err = 0;
 	M0_ENTRY();
 
 	switch (m0_fom_phase(fom)) {
@@ -250,17 +250,14 @@ static int addb_pfom_fo_tick(struct m0_fom *fom)
 		M0_LOG(M0_DEBUG, "post");
 		m0_reqh_stats_post_addb(reqh);
 
-#if 0
-		err = m0_addb_monitor_summaries_post(svc->as_reqhs.rs_reqh,
-						     pfom);
-
+		if (reqh->rh_addb_monitoring_ctx.amc_stats_ep != NULL)
+			err = m0_addb_monitor_summaries_post(reqh, pfom);
 		/**
 		 * In case of summaries posting failure, just log error.
 		 * We do not terminate the fom.
 		 */
 		if (err != 0)
 			M0_LOG(M0_ERROR, "addb summary posting failed");
-#endif
 
 /** Only needed for stobsink, so should not be called in kernel */
 #ifndef __KERNEL__
-- 
1.8.3.2

